### ISSUE TITLE
DisplayActionSheetBehavior crashes when click at outside of dialog on Android.

### ISSUE THREAD

**Issue by noxi515** [nuitsjp/BehaviorsForXamarinForms#3]
DisplayActionSheetBehavior always crashes when i click at outside of the dialog.  
When i click outside, `Page.DisplayActionSheet` returns null value. Behavior does not consider that.

Env:

* Android
* Xamarin.Forms 2.4.0.280
* Xamarin.Android 7.1
* Xamarin.Forms.BehaviorsPack 1.3.0

**Comment by nuitsjp** [comment]
I tried to investigate but it did not reproduce.
I have two requests.

1. If you run BehaviorsSampleApp.sln from this repository in your environment, please confirm whether to reproduce
2. Can you provide code that can be reproduced?

**Comment by noxi515** [comment]
I made a reproducable code at repository [here](https://github.com/noxi515/xamarin-forms-behaviorspack-issue3).  

Step:

1. Create a solution by VS2017.3.5 Enterprise.
1. Add a Xamrin.Forms PCL project.
1. Update Xamarin.Forms to 2.4.0.282.
1. Add NuGet dependency Xamarin.Forms.BehaviorsPack 1.3.0.
1. Add a button with DisplayActionSheetBehavior in MainPage.xaml.
1. Run Android project in Android Emulator.
1. Show action sheet dialog.
1. Click outside dialog.

![xamarin-issue3](https://user-images.githubusercontent.com/1465111/31112745-8567536a-a850-11e7-9e9c-725f73babb79.png)


Environment details:

* Windows 10 Enterprise (1703 x64)
* Visual Studio 2017 Enterprise (15.3.5)
* Xamarin 4.7.9.45
* Xamain.Android SDK 7.4.5.1
* Android Emulator (with Intel HAXM)
  * Android 5.0 (Google APIs Intel x86 Atom System Image rev.23)
  * Android 6.0 (Google APIs Intel x86 Atom System Image rev.23)
  * Android 7.0 (Google Play Intel x86 Atom System Image rev.16)



StackTrace below.

```
UNHANDLED EXCEPTION:
System.InvalidOperationException: Sequence contains more than one matching element
  at System.Linq.Enumerable.Single[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x0004a] in <b5bd9d990a0b4733885e90ca5ec6c0fb>:0 
  at Xamarin.Forms.BehaviorsPack.DisplayActionSheetBehavior+<OnReceived>d__62.MoveNext () [0x001e1] in <a69417759d53480d93a9c171675a6472>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <896ad1d315ca4ba7b117efb8dacaedcf>:0 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in <896ad1d315ca4ba7b117efb8dacaedcf>:0 
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <2e14bb2dd93a405e81838369ed72695b>:0 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <2e14bb2dd93a405e81838369ed72695b>:0 
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <2e14bb2dd93a405e81838369ed72695b>:0 
  at (wrapper dynamic-method) System.Object:1eb7b53d-f3e0-490e-bb0b-143d5c7bc307 (intptr,intptr)
```

**Comment by nuitsjp** [comment]
I appreciate your great contribution.
Fixed this Issue and released 1.3.1.
Thank you very much for helping me.
